src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -223,20 +241,12 @@ private bool getReadyToRunHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, ref
                         {
                             helperArg = new MethodWithToken(methodArg, new ModuleToken(_tokenContext, pResolvedToken.token));
                         }
-                        TypeDesc contextType;
-                        if (pGenericLookupKind.runtimeLookupKind == CORINFO_RUNTIME_LOOKUP_KIND.CORINFO_LOOKUP_THISOBJ)
-                        {
-                            contextType = methodFromContext(pResolvedToken.tokenContext).OwningType;
-                        }
-                        else
-                        {
-                            contextType = null;
-                        }
+                        MethodContext methodContext = new MethodContext(methodFromContext(pResolvedToken.tokenContext), new IntPtr(pResolvedToken.tokenContext));
```

Comment:
How does this work? `pResolvedToken.tokenContext` is just a cookie that `ObjectToHandle` generates in JitInterface. This cookie has no meaning outside JitInterface.

Reply:
> This cookie has no meaning outside JitInterface

(In particular, there's no guarantees about it's uniqueness or equality with other cookies from different method compilations.)

Reply:
Hmm, I used the tokenContext because JanK said Crossgen uses it as the lookup key for generic lookup cells too. I can limit the context to just the method but that's going to be susceptible to the same version resiliency problem as we discussed in Redmond with JanK. I'd be much happier to keep track of the context in a more reliable manner, please let me know if you're aware of any. AFAIK today JIT interface doesn't seem to provide any token information for the context method.

Reply:
The cookie can be decoded to a method or a type using `methodFromContext` or `typeFromContext`. I'm pretty lost in how this code fits together, but using a type system entity instead of a cookie that is only meaningful within a single `CorInfoImpl` instance seems like a better plan. The cookies escaping `CorInfoImpl` could lead to some very subtle bugs once we go multithreaded and have multiple `CorInfoImpl` instances in the system.

Reply:
Michal is right. The IntPtr field in MethodContext is wrong - it is a cookie that is valid only for the single method code generation. It should be the type system entity that it maps to instead.

Note that it is different from crossgen. In crossgen, `tokenContext` is actual `MethodDesc*` or `MethodTable*` that is valid for the duration of the compilation process.